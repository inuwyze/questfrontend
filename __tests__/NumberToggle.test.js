import React from 'react';
import renderer from 'react-test-renderer';
import NumberToggle from '../components/numberToggle';
import { render, fireEvent } from '@testing-library/react-native';
describe('Functional Testing numberToggle', () => {
  const onMockPress=jest.fn()
  it('check if function is called', () => {
    
    const {getByText}=render(<NumberToggle
    number={5}
    setNumber={onMockPress}
    />)
    fireEvent.press(getByText('+'))
    expect(onMockPress).toHaveBeenCalled()
  });
  it('check if increased number is passed to setNumber', () => {
    const {getByText}=render(<NumberToggle
    number={5}
    setNumber={onMockPress}
    />)
    fireEvent.press(getByText('+'))
    expect(onMockPress).toHaveBeenCalledWith(6)
  });
  it('check if decreased number is passed to setNumber', () => {
    const {getByText}=render(<NumberToggle
    number={5}
    setNumber={onMockPress}
    />)
    fireEvent.press(getByText('-'))
    expect(onMockPress).toHaveBeenCalledWith(4)
  });
  it('NumberToggle renders correctly', () => {
    const tree = renderer.create(<NumberToggle 
        number={5}
        setNumber={onMockPress}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

});