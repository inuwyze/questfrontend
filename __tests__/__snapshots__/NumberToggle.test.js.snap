// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Functional Testing numberToggle NumberToggle renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "width": 280,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "black",
        "borderColor": "orange",
        "borderRadius": 10,
        "borderWidth": 1,
        "flex": 1,
        "justifyContent": "center",
        "marginRight": 8,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 24,
        }
      }
    >
      -
    </Text>
  </View>
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "orange",
          "borderRadius": 10,
          "borderWidth": 1,
          "flex": 3,
          "padding": 10,
          "width": 280,
        },
      ]
    }
    underlineColorAndroid="transparent"
    value="5"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "black",
        "borderColor": "orange",
        "borderRadius": 10,
        "borderWidth": 1,
        "flex": 1,
        "justifyContent": "center",
        "marginLeft": 8,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 24,
        }
      }
    >
      +
    </Text>
  </View>
</View>
`;
